bodhi/server/models.py

```diff
@@ -1924,11 +1925,23 @@ def get_test_gating_info(self):
 
     def update_test_gating_status(self):
         """Query Greenwave about this update and set the test_gating_status as appropriate."""
-        decision = self.get_test_gating_info()
+        try:
+            decision = self.get_test_gating_info()
+            decision['greenwave_success'] = True
+        except (requests.exceptions.Timeout, RuntimeError) as e:
+            log.error(str(e))
+            # Greenwave frequently returns 500 response codes. When this happens, we do not want
+            # to block updates from proceeding, so we will consider this condition as having the
+            # policy satisfied. We will use the Exception as the summary so we can mark the status
+            # as ignored for the record.
+            decision = {'policies_satisfied': True, 'summary': str(e), 'greenwave_success': False}
         if decision['policies_satisfied']:
-            # If an unrestricted policy is applied and no tests are required
-            # on this update, let's set the test gating as ignored in Bodhi.
-            if decision['summary'] == 'no tests are required':
+            if not decision['greenwave_success']:
+                # Greenwave failed to respond, so let's mark this as ignored.
+                self.test_gating_status = TestGatingStatus.ignored
```

Comment:
Do we maybe want a new status for this, so people can distinguish between ignored due to Greenwave not configured vs `TestGatingStatus.GreenwaveFailed`?

Reply:
We could do that, but it would require a database migration. Do note that we are also recording the error message in the ```greenwave_summary_string``` below.

I don't mind adding the migration if you prefer that option. I agree that it would be cleaner. However, I won't be able to add that today (I can do it tomorrow, I just need to stop working due to evening plans.)

Reply:
After thinking about it some more, I do like your suggestion of extending the status to cover this. It will help us be more explicit about what is happening than the summary string will be.

I will add that tomorrow and then we can proceed.

Reply:
@puiterwijk alright, I have a migration in place now that adds a new status to track when Greenwave failed.

Reply:
Oooh, but now I need to make sure that Bodhi does the right thing with this new state…

Reply:
OK, I've got a local patch to make Bodhi care about the new state, but I need tests and I need to leave. I'll get that posted up tomorrow.